File: Android/tuilivekit/component/barrage/src/main/java/com/trtc/uikit/component/barrage/service/DataReporter.java
package com.trtc.uikit.component.barrage.service;

import android.util.Log;

import com.tencent.cloud.tuikit.engine.room.TUIRoomEngine;

import org.json.JSONException;
import org.json.JSONObject;

public class DataReporter {
    public static void reportEventData(int eventKey) {
        try {
            JSONObject params = new JSONObject();
            params.put("key", eventKey);
            JSONObject jsonObject = new JSONObject();
            jsonObject.put("api", "KeyMetricsStats");
            jsonObject.put("params", params);
            String json = jsonObject.toString();
            TUIRoomEngine.callExperimentalAPI(json);
            Log.i("DataReporter", "reportEventData:[json:" + json + "]");
        } catch (JSONException e) {
            Log.e("DataReporter", "reportEventData:[e:" + e.getMessage() + "]");
        }
    }
}

File: iOS/TUILiveKit/Sources/Common/Log/DataReporter.swift
//
//  DataReporter.swift
//  TUILiveKit
//
//  Created by adamsfliu on 2024/6/13.
//

import Foundation
import RTCRoomEngine

public class DataReporter {
    
    public enum ComponentType: Int {
        case liveRoom = 21
        case voiceRoom = 22
    }
    
    private static let framework: Int = 1
    private static let language: Int = 3
    public static var componentType: ComponentType = .liveRoom
    static func reportFramework() {
        let apiParams: [String : Any] = [
            "api": "setFramework",
            "params": [
              "framework": framework,
              "component": componentType.rawValue,
              "language": language,
            ],
          ]
        callExperimentalAPI(params: apiParams)
    }
    
    public static func reportEventData(eventKey: Int) {
        let apiParams: [String: Any] = [
            "api": "KeyMetricsStats",
            "params": [
                "key": eventKey,
            ],
        ]
        callExperimentalAPI(params: apiParams)
    }
    
    private static func callExperimentalAPI(params: [String : Any]) {
        do {
            let jsonData = try JSONSerialization.data(withJSONObject: params, options: .prettyPrinted)
            if let jsonString = String(data: jsonData, encoding: .utf8) {
                print(jsonString)
                TUIRoomEngine.callExperimentalAPI(jsonStr: jsonString)
            } else {
                print("Error converting JSON data to string")
            }
        } catch {
            print("Error converting dictionary to JSON: \(error.localizedDescription)")
        }
    }
}
